plugin/trino-hive/src/main/java/io/trino/plugin/hive/procedure/FlushMetadataCacheProcedure.java

```diff
@@ -144,6 +145,11 @@ private void doFlushMetadataCache(ConnectorSession session, Optional<String> sch
                 partitionColumns.size() == partitionValues.size(),
                 "Parameters partition_column and partition_value should have same length");
 
+        Optional<CachingHiveMetastore> cachingHiveMetastore = this.cachingHiveMetastore;
+        if (hiveMetadataFactory instanceof ImpersonationCachingHiveMetastoreFactory impersonationCachingHiveMetastoreFactory) {
+            cachingHiveMetastore = Optional.of((CachingHiveMetastore) impersonationCachingHiveMetastoreFactory.createMetastore(Optional.of(session.getIdentity())));
```

Comment:
maybe we could move the logic at the beginning of the method, after determine the metastore we can keep the remaining logic.